pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -95,9 +95,9 @@ public void runJob() {
 
               if (!result.isEndOfStreamBlock()) {
                 // not complete, needs to re-register for scheduling
-                register(operatorChain);
+                register(operatorChain, false);
```

Comment:
what if an operatorChain is reregistered but not hooked with any mailbox?

Reply:
that should never happen with the changes to the operator chain, the operators process everything that they can process until all inputs available. in the case of those not hooked with any mailbox they will process everything and the operator will finish.

Reply:
got it. e.g. an operator either waits on a mailbox, or should finish entirely (e.g. must return an EOS block at the end)